### DISCUSSION TITLE
Send Language in options

### DISCUSSION THREAD

**Discussion by VedikaGujar-GSLab** [dodona-edu/dolos#1105] [Q&A] [answered]
Hi All,

I want to do plagiarism check for html, sql language, I have installed respective tree sitter library.
But it still gives me error below error
```
"Error: Could not detect language based on extension (.html)."
"Error: Could not detect language based on extension (.sql)."

"LanguageError: Could not detect language based on extension (.html).
    at LanguagePicker.detectLanguage"
 ```

I also tried sending the language using options
```js
const instance = new dolos.Dolos();
instance.options.language = "html";
const report = instance.analyzePaths(files);
```

but it still doesn't work, can you please help me with the same.

**Comment by VedikaGujar-GSLab** [comment, marked as answer]
I further modified the above code and it started working, please help to figure out if its the right solution
```js
const customOptions = {
   language: "html"
 }

const options = new dolos.Options(customOptions);
const instance = new dolos.Dolos(options);
const report = instance.analyzePaths(files);
```

**Reply by rien** [reply]
Yes, this indeed how you would specify the programming language for a parser that is not installed with Dolos by default.

We have a [`LanguagePicker`](https://github.com/dodona-edu/dolos/blob/main/lib/src/lib/util/language.ts#L99) class that currently only detects languages based on extension if they are installed with Dolos by default. We could however change that. Would you prefer this to be the case?
